lib/Doctrine/ORM/Mapping/Driver/AnnotationDriver.php

```diff
@@ -712,18 +715,7 @@ private function joinColumnToArray(Mapping\JoinColumn $joinColumn): array
      * Parse the given Column as array
      *
      * @return mixed[]
-     * @psalm-return array{
-     *                   fieldName: string,
-     *                   type: mixed,
-     *                   scale: int,
-     *                   length: int,
-     *                   unique: bool,
-     *                   nullable: bool,
-     *                   precision: int,
-     *                   options?: mixed[],
-     *                   columnName?: string,
-     *                   columnDefinition?: string
-     *               }
+     * @psalm-return FieldMapping
```

Comment:
this is a BC break IMHO, previous type stats that `fieldName`, `type`, `scale`, `length`, `unique`, `nullable`, and `precision` are always available, while the new type states that they might not exist in the returned array.

Reply:
Oh right, didn't catch that, why did you change this @laryjulien ?

Reply:
I have changed my PR to only use FieldMapping type in the initial context of issue.
I suggest to open an other PR to use this type in a wider context. Indeed, there are small variations on field mapping definition which appear to be more complex than I thought at first glance.